src/overlays/actors/ovl_En_Horse_Normal/z_en_horse_normal.c

```diff
@@ -21,45 +48,635 @@ const ActorInit En_Horse_Normal_InitVars = {
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Normal/func_80A6B250.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Horse_Normal/func_80A6B30C.s")
+extern AnimationHeader D_06000608;
+extern AnimationHeader D_06000C20;
+extern AnimationHeader D_060013A8;
+extern AnimationHeader D_06001A1C;
+extern AnimationHeader D_06002458;
+extern AnimationHeader D_060035D4;
+extern AnimationHeader D_06004580;
+extern AnimationHeader D_06004C20;
+extern AnimationHeader D_060054BC;
+
+extern SkeletonHeader D_06009FAC;
+
+static AnimationHeader* sAnimations[] = {
+    &D_06004580, &D_06004C20, &D_060035D4, &D_06002458, &D_060054BC, &D_06001A1C, &D_06000608, &D_06000C20, &D_060013A8,
+};
+
+static ColliderCylinderInit sCylinderInit1 = {
+    { COLTYPE_UNK10, 0x00, 0x00, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+    { 40, 100, 0, { 0, 0, 0 } },
+};
+
+static ColliderCylinderInit sCylinderInit2 = {
+    { COLTYPE_UNK10, 0x00, 0x00, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+    { 60, 100, 0, { 0, 0, 0 } },
+};
+
+static ColliderJntSphItemInit sJntSphItemsInit[] = {
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+        { 11, { { 0, 0, 0 }, 20 }, 100 },
+    },
+};
+
+static ColliderJntSphInit sJntSphInit = {
+    { COLTYPE_UNK10, 0x00, 0x00, 0x39, 0x10, COLSHAPE_JNTSPH },
+    1,
+    sJntSphItemsInit,
+};
+
+static CollisionCheckInfoInit sColChkInfoInit = { 10, 35, 100, 0xFE };
+
+// Unused
+static EnHorseNormalUnkStruct1 D_80A6D428[] = {
+    { { 1058, 1, 384 }, 7 },    { { 1653, 39, -381 }, 6 }, { { 1606, 1, -1048 }, 6 }, { { 1053, 1, -1620 }, 6 },
+    { { -1012, 1, -1633 }, 7 }, { { -1655, 1, -918 }, 6 }, { { -1586, 1, -134 }, 6 }, { { -961, 1, 403 }, 7 },
+};
+
+// Unused
+static EnHorseNormalUnkStruct2 D_80A6D468 = { ARRAY_COUNT(D_80A6D428), D_80A6D428 };
+
+// Unused
+static EnHorseNormalUnkStruct1 D_80A6D470[] = {
+    { { 88, 0, 2078 }, 10 },       { { 2482, 376, 4631 }, 7 },    { { 2228, -28, 6605 }, 12 },
+    { { 654, -100, 8864 }, 7 },    { { -297, -500, 10667 }, 12 }, { { -5303, -420, 10640 }, 10 },
+    { { -6686, -500, 7760 }, 10 }, { { -5260, 100, 5411 }, 7 },   { { -3573, -269, 3893 }, 10 },
+};
+
+// Unused
+static EnHorseNormalUnkStruct2 D_80A6D4B8 = { ARRAY_COUNT(D_80A6D470), D_80A6D470 };
+
+void func_80A6B250(EnHorseNormal* this) {
+    static s32 D_80A6D4C0[] = { 0, 16 };
+
+    if (D_80A6D4C0[this->unk_200] < this->skin.skelAnime.animCurrentFrame &&
+        ((this->unk_200 != 0) || !(D_80A6D4C0[1] < this->skin.skelAnime.animCurrentFrame))) {
+        Audio_PlaySoundGeneral(NA_SE_EV_HORSE_WALK, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+        this->unk_200 += 1;
+        if (this->unk_200 >= 2) {
```

Comment:
```suggestion
        this->unk_200++;
        if (this->unk_200 >= ARRAY_COUNT(D_80A6D4C0)) {
```
since it indexes into `D_80A6D4C0` the 2 must be the length